modules/javafx.swing/src/main/java/javafx/embed/swing/InputMethodSupport.java

```diff
@@ -56,42 +58,64 @@ public InputMethodRequestsAdapter(javafx.scene.input.InputMethodRequests fxReque
             this.fxRequests = fxRequests;
         }
 
+        private Point2D pointValue;
+        private int     intValue;
+        private String  stringValue;
```

Comment:
Using instance fields like this is not thread-safe in general (although unlikely to cause problems in practice for this specific case). I recommend either creating a utility method that returns the result of a `Future` (see PR #1321 ) or changing these to local `AtomicObject<Type>` / `AtomicInteger` (or `Type[]` `int[]`) variables.

Reply:
Using instance fields here does look clunky. There's no thread-safety issue because the JFXPanel code creates a new adapter for every request so I decided to avoid the overhead of creating local arrays (apologies for the micro-optimizing). I will switch to local array variables if you want.

For the same reason I don't think the `getInputMethodRequests` call added to line 1028 in PR #1169 is necessary. The adapter it creates is discarded so the only side-effect is calling the scene's `getInputMethodRequests` before the requests are needed. I can't think of any reason this front-loading would be necessary.